rasa/server.py

```diff
@@ -776,6 +797,7 @@ async def trigger_intent(request: Request, conversation_id: Text) -> HTTPRespons
     @app.post("/conversations/<conversation_id:path>/predict")
     @requires_auth(app, auth_token)
     @ensure_loaded_agent(app)
+    @ensure_conversation_exists(app)
     async def predict(request: Request, conversation_id: Text) -> HTTPResponse:
         try:
```

Comment:
out of the scope of your PR / optional: We should actually use the `lockstore` here as `processor.predict_next` persists the updated tracker.